src/workerd/api/crypto/dh.c++

```diff
@@ -147,67 +151,80 @@ bool CryptoImpl::DiffieHellmanHandle::Init(
                   Error, "DiffieHellman init failed: could not set keys");
       bn_g.release();
       bn_p.release();
-      return VerifyContext();
+      return kj::mv(dh);
     }
   }
-
   KJ_UNREACHABLE;
 }
 
-void CryptoImpl::DiffieHellmanHandle::setPrivateKey(kj::Array<kj::byte> key) {
+void zeroPadDiffieHellmanSecret(size_t remainder_size,
+                                unsigned char* data,
+                                size_t prime_size) {
+  // DH_size returns number of bytes in a prime number.
+  // DH_compute_key returns number of bytes in a remainder of exponent, which
+  // may have less bytes than a prime number. Therefore add 0-padding to the
+  // allocated buffer.
+  if (remainder_size != prime_size) {
+    KJ_ASSERT(remainder_size < prime_size);
+    const size_t padding = prime_size - remainder_size;
+    memmove(data + padding, data, remainder_size);
```

Comment:
I don't think this is constant-time. Does it matter in this context?

Reply:
I wouldn't consider it critical right now. If we didn't have spectre mitigations in place then it might be an issue but as it is I think this should be fine.

Reply:
Hmm. Makes me a bit queasy but I guess it's fine.